File: src/main/java/dev/klomptech/jbark/interpreter/StdinReader.java
package dev.klomptech.jbark.interpreter;

import dev.klomptech.jbark.errors.BarkError;
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public final class StdinReader {

  private StdinReader() {}

  public static BarkValue read(final int line) throws BarkError {
    try {
      String input = new BufferedReader(new InputStreamReader(System.in)).readLine();
      if (input == null) {
        return new BarkValue.BarkNull();
      }
      return BarkValue.of(input);
    } catch (IOException _) {
      throw new BarkError(line, "The human went quiet. Couldn't catch anything from stdin.");
    }
  }
}

File: pbark/pbark/interpreter/stdin_reader.py
from __future__ import annotations

import sys

from pbark.errors import BarkError
from pbark.interpreter import bark_value as bv


class StdinReader:
    @staticmethod
    def read(line: int) -> bv.BarkValue:
        try:
            input_line = sys.stdin.readline()
            if input_line == "":
                return bv.BarkNull()
            return bv.of(input_line.rstrip("\n"))
        except OSError as exc:
            raise BarkError(line, "The human went quiet. Couldn't catch anything from stdin.") from exc
